packages/tinacms/src/internalClient/index.ts

```diff
@@ -520,12 +645,67 @@ mutation addPendingDocumentMutation(
     }
   }
 
+  async waitForIndexStatus({ ref }: { ref: string }) {
+    const result = await asyncPoll(
+      async (): Promise<AsyncData<any>> => {
+        try {
+          const result = await this.getIndexStatus({ ref })
+          if (
+            !(result.status === 'inprogress' || result.status === 'unknown')
+          ) {
+            return Promise.resolve({
+              done: true,
+              data: result,
+            })
+          } else {
+            return Promise.resolve({
+              done: false,
+            })
+          }
+        } catch (err) {
+          return Promise.reject(err)
+        }
+      },
+      // interval is 1000ms
+      1000, // interval
+      //  timeout is 10 min
+      600000 // timeout
```

Comment:
after thinking about this some more, I think given the lambda timeout is 15 mins, that is probably a good number here (so 900000)